lib/media/drm_engine.js

```diff
@@ -727,11 +738,19 @@ shaka.media.DrmEngine = class {
           // MediaKeySystemConfiguration
           config['drmInfos'].push(info);
 
+          if (info.initData && info.initData.length) {
+            config.initDataTypes = [
+              ...new Set(info.initData.map(({initDataType}) => initDataType)),
+            ];
+          }
+
           if (info.distinctiveIdentifierRequired) {
             config.distinctiveIdentifier = 'required';
           }
+
           if (info.persistentStateRequired) {
             config.persistentState = 'required';
+            config.sessionTypes = ['persistent-license'];
```

Comment:
This isn't always true, we can still require persistent state for `temporary` licenses.  IIRC, some people use this with Widevine to force the CDM to send a client ID.  This also has the problem of the license not getting destroyed.  Once we stop playing, the `persistent-license` session will remain on disk; we'd need to explicitly call `session.remove()` to remove from disk when we stop playing.